src/cmds/qdel.c

```diff
@@ -91,13 +91,13 @@ process_deljobstat(char *clusterid, struct batch_deljob_status **list, svr_jobid
 
 			/* Check if job was moved to a remote cluster */
 			if (locate_job(p_delstatus->name, clusterid, rmt_server)) {
-				if (add_jid_to_list_by_name(p_delstatus->name, rmt_server, rmtlist) != 0)
+				if (add_jid_to_list_by_name(strdup(p_delstatus->name), rmt_server, rmtlist) != 0)
 					return pbs_errno;
 				else {	/* Job found on remote server, let's remove it from error list */
 					if (prev != NULL)
-						*list = next;
-					else
 						prev->next = next;
+					else
```

Comment:
:) how did we miss this

Reply:
I can't wait until we can use std::list<> for something like this.

Reply:
haha :)